cdap-common/src/main/java/co/cask/cdap/common/guice/LocationRuntimeModule.java

```diff
@@ -80,6 +80,26 @@ private RootLocationFactory providesRootLocationFactory(CConfiguration cConf) {
     }
   }
 
+  public static UserGroupInformation getUserGroupInformation(CConfiguration cConf) {
```

Comment:
You can piggy back on the `co.cask.cdap.common.security.YarnTokenUtils` class for now.